riot/riot.ml

```diff
@@ -40,6 +40,20 @@ let run ?(rnd = Random.State.make_self_init ()) ?workers main =
   Log.debug (fun f -> f "Riot runtime shutdown");
   Stdlib.exit pool.status
 
+let default_on_error error =
```

Comment:
we can constrain the error here instead to make sure that if the default function is used, then the only error that can be returned is a `` `Msg "error" ``.

```suggestion
let default_on_error (error: [| `Msg of string ]) =
```

If people want custom errors, they need to pass in a custom `on_error` function.

Reply:
Should be resolved by [ef0393f](https://github.com/riot-ml/riot/pull/61/commits/ef0393f223d696772a9178fc4f9d0c9a16195782)